ONSdigital/dp-static-file-publisher
# Dis 838 fix local healthcheck issue with dp static file publisher

### What

[DIS-838](https://jira.ons.gov.uk/browse/DIS-838) Fix local healthcheck issue with dp-static-file-publisher

When dp-static-file-publisher is run locally, S3 functionality is emulated using [localstack](https://www.localstack.cloud/). The public and private S3 buckets required by dp-static-file-publisher are created on localstack, and dp-static-file-publisher runs a healthcheck to confirm these buckets exist and the app can connect to them. This healtcheck works fine when connecting to the real S3, but was failing when connecting to localstack.

dp-static-file-publisher was configured to use two versions of the [dp-s3](https://github.com/ONSdigital/dp-s3) S3Client, v1 and v2. The v1 client was used for the private and public bucket healthchecks. We found that the v1 client was not instantiated with the correct `Endpoint` and `S3ForcePathStyle` configuration to allow it to connect properly to the localstack S3 buckets. This was causing the `BucketRegionError`.

Fixes:
- Updated the S3 clients used for the healthchecks in `service/initialise.go` to use the dp-s3 AWS SDK V2 client, which configures the client to work with localstack correctly on instantiation
- Updated tests and dependencies to use dp-s3 v2 only

### How to review

* Run the app locally (e.g. using the [dataset-catalogue](https://github.com/ONSdigital/dp-compose/tree/main/v2/stacks/dataset-catalogue) stack in dp-compose)
* View the healthcheck outputs at `http://localhost:24900/health`
* Confirm healthcheck output reports that both `S3 Private` and `S3 Public` buckets have the `S3 is healthy` message
